lib/galaxy/tool_util/client/staging.py

```diff
@@ -178,9 +179,7 @@ def _attach_file(upload_payload: Dict[str, Any], uri: str, index: int = 0) -> No
                 file_type = upload_target.properties.get("filetype", None) or DEFAULT_FILE_TYPE
                 dbkey = upload_target.properties.get("dbkey", None) or DEFAULT_DBKEY
                 upload_payload = _upload_payload(
-                    history_id,
-                    file_type=file_type,
-                    to_posix_lines=dbkey,
+                    history_id, file_type=file_type, to_posix_lines=dbkey, hashes=upload_target.properties.get("hashes")
```

Comment:
I think you need to add processing of `hashes` to `_upload_payload()` below.

Reply:
I've dropped that branch, I think we should focus on fetch